### ISSUE TITLE
Why aren't there "aria-label" in components

### ISSUE THREAD

**Issue by dawangdw** [carbon-design-system/carbon-components-react#766]
## Detailed description

> Describe in detail the issue you're having.

Because there is no "aria-label" in inner element of component(e.g. [CodeSnippet](https://github.com/carbon-design-system/carbon-components-react/blob/master/src/components/CodeSnippet/CodeSnippet.js#L19)), I am not able to pass accessibility check.

> Is this a feature request (new component, new icon), a bug, or a general issue?

General issue.

> Is this issue related to a specific component?

So far, I find two components having this kind of problem. They are [CodeSnippet](https://github.com/carbon-design-system/carbon-components-react/blob/master/src/components/CodeSnippet/CodeSnippet.js#L19) and [AccordionItem](https://github.com/carbon-design-system/carbon-components-react/blob/master/src/components/AccordionItem/AccordionItem.js#L76).

> What did you expect to happen? What happened instead? What would you like to see changed?

I expect we include "aria-label" in inner element of components, or maybe provide a way to customize.

> What browser are you working in?

Chrome.

> What version of the Carbon Design System are you using?

```
    "carbon-components": "^8.1.11",
    "carbon-components-react": "^5.32.0",
```

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

ASAP.

**Comment by joshblack** [comment]
Hi @dawangdw! 👋 Thanks so much for taking the time to make this issue 🙂

With respect to accessibility labels, we definitely try to provide them when appropriate, especially to help you pass accessibility checks!

If you could share the issue that you're encountering, would love to help figure out a way to add support in for the accessibility labels you might need 😄

**Comment by dawangdw** [comment]
@joshblack Thanks for responding so fast!

Here are my problems:

1. For [CodeSnippet](https://github.com/carbon-design-system/carbon-components-react/blob/master/src/components/CodeSnippet/CodeSnippet.js), I need to add "aria-label" in [this div](https://github.com/carbon-design-system/carbon-components-react/blob/master/src/components/CodeSnippet/CodeSnippet.js#L19)(like `<div role="textbox" tabIndex={0} className="bx--snippet-container" aria-label="text box">`), then I can pass the accessibility check.

Here is corresponding check report:
```
...
"rpt.g1110.widgetLabels": "A WAI-ARIA widget must have an accessible name (see message help text for details).",
...
 "issues": [
    {
        "bounds": {
            "height": 80,
            "left": 325,
            "top": 822,
            "width": 246
        },
        "help": "idhi_accessibility_check_g1110.html",
        "level": "violation",
        "messageCode": "rpt.g1110.widgetLabels",
        "msgArgs": [],
        "ruleId": "1110",
        "severityCode": "eISHigh",
        "snippet": "<div class=\"bx--snippet-container\" tabindex=\"0\" role=\"textbox\">",
        "xpath": "/html[1]/body[1]/div[2]/div[1]/main[1]/section[1]/div[1]/article[1]/ul[1]/li[1]/div[1]/div[1]/ul[1]/li[2]/div[1]/div[2]/div[1]/div[1]"
    },
...
```


2. For [AccordionItem](https://github.com/carbon-design-system/carbon-components-react/blob/master/src/components/AccordionItem/AccordionItem.js), in this [button](https://github.com/carbon-design-system/carbon-components-react/blob/master/src/components/AccordionItem/AccordionItem.js#L76) it reports violation as well, which I don't know how to handle it yet. (I am new to accessibility.)

Here is the corresponding check report:
```
...
"haac.g1142.AriaSemantics": "The WAI-ARIA role(s) {0} do not meet the default implicit ARIA semantic(s) and restriction(s) for the element {1}. If specified allowed roles must be {2}.",
...
{
    "bounds": {
        "height": 36,
        "left": 240,
        "top": 1258,
        "width": 685
    },
    "help": "idhi_accessibility_check_g1142.html",
    "level": "violation",
    "messageCode": "haac.g1142.AriaSemantics",
    "msgArgs": [
        "tab",
        "button",
        "button, link, menuitem, menuitemcheckbox, menuitemradio, radio"
    ],
    "ruleId": "1142",
    "severityCode": "eISMedium",
    "snippet": "<button role=\"tab\" class=\"bx--accordion__heading\" type=\"button\">",
    "xpath": "/html[1]/body[1]/div[2]/div[1]/main[1]/section[1]/div[1]/article[1]/ul[1]/li[4]/button[1]"
},
...
```

**Comment by joshblack** [comment]
Closing this for our internal audit issue. Look for support for this in an upcoming release!
